sway-lib-std/src/inputs.sw

```diff
@@ -156,3 +175,57 @@ pub fn input_count() -> u8 {
         },
     }
 }
+
+////////////////////////////////////////
+// Message Inputs
+////////////////////////////////////////
+
+/// Get the message id of the input message at `index`.
+pub fn input_message_msg_id(index: u64) -> b256 {
+    __gtf::<b256>(index, GTF_INPUT_MESSAGE_MESSAGE_ID)
+}
+
+/// Get the sender of the input message at `index`.
+pub fn input_message_sender(index: u64) -> Address {
+    ~Address::from(__gtf::<b256>(index, GTF_INPUT_MESSAGE_SENDER))
+}
+
+/// Get the recipient of the input message at `index`.
+pub fn input_message_recipient(index: u64) -> Address {
+    ~Address::from(__gtf::<b256>(index, GTF_INPUT_MESSAGE_RECIPIENT))
+}
+
+/// Get the nonce of input message at `index`.
+pub fn input_message_nonce(index: u64) -> b256 {
+    __gtf::<b256>(index, GTF_INPUT_MESSAGE_NONCE)
+}
+
+/// Get the witness index of the input messasge at `index`.
+pub fn input_message_witness_index(index: u64) -> u64 {
+    __gtf::<u64>(index, GTF_INPUT_MESSAGE_WITNESS_INDEX)
+}
+
+/// Get the length of the input message at `index`.
+pub fn input_message_data_length(index: u64) -> u64 {
+    __gtf::<u64>(index, GTF_INPUT_MESSAGE_DATA_LENGTH)
+}
+
+/// Get the predicate length of the input message at `index`.
+pub fn input_message_predicate_length(index: u64) -> u64 {
+    __gtf::<u64>(index, GTF_INPUT_MESSAGE_PREDICATE_LENGTH)
+}
+
+/// Get the predicate data length of the input message at `index`.
+pub fn input_message_predicate_data_length(index: u64) -> u64 {
+    __gtf::<u64>(index, GTF_INPUT_MESSAGE_PREDICATE_DATA_LENGTH)
+}
+
+/// Get the data of the input message at `index`.
+pub fn input_message_data<T>(index: u64) -> T {
```

Comment:
This should include an `offset: u64` field so individual reads in a larger data set can be performed

Reply:
@pixelcircuits Would the user need to pass in a `start_offset` and a `length` for the data they actually care about then?

Reply:
No, I've been using my own version of this function like this: 
```
pub fn input_message_data<T>(index: u64, offset: u64) -> T {
    read::<T>(__gtf::<u64>(index, GTF_INPUT_MESSAGE_DATA) + offset)
}
```

It let's me parse data from a message like this:
```
// Parse the message data
let contract_id: b256 = input_message_data(msg_idx, 0);
storage.data1 = ~ContractId::from(contract_id);

let num: u64 = input_message_data(msg_idx, 32);
storage.data2 = num;

let big_num: b256 = input_message_data(msg_idx, 32 + 8);
storage.data3 = big_num;

let address: b256 = input_message_data(msg_idx, 32 + 8 + 32);
storage.data4 = ~Address::from(address);
```

It seems to work just fine.